### PR TITLE
Add support for image sorting for all options

### LINKED ISSUE #6670: podman images -q doesn't sort
I am fairly certain that this is causing the CI flakes in `podman rmi`. I unflaked with using `--format` in https://github.com/containers/libpod/pull/6669 but the underlying problem persists. We also need to add tests to make sure that -q sorts correctly.

```
libpod (unflake-rmi) $ ./bin/podman images -a --sort created --format "{{.ID}}"
45f80ba3db88
a24bb4013296

libpod (unflake-rmi) $ ./bin/podman images -a --sort created -q
a24bb4013296
45f80ba3db88
```

### PR DESCRIPTION
**Pull request by jwhonce**
* Sort images for -q option, removing duplicate id's

Fixes #6670

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM 
assuming happy tests

**Comment by rhatdan** [comment]
Can you fix your commit message to something a little more descriptive.  This message will be useless in the future.

File: cmd/podman/images/list.go [hunk 1]
```diff
@@ -109,16 +109,18 @@ func images(cmd *cobra.Command, args []string) error {
 }
 
 func writeID(imageS []*entities.ImageSummary) error {
-	var ids = map[string]struct{}{}
-	for _, e := range imageS {
-		i := "sha256:" + e.ID
-		if !listFlag.noTrunc {
-			i = fmt.Sprintf("%12.12s", e.ID)
+	imgs := sortImages(imageS)
```
Comment (vrothberg) [inline cmd/podman/images/list.go:112]: I think we should sort the `imageS` before at the caller.

**Comment by vrothberg** [comment]
I noticed that `--format json` doesn't sort either, but already in v1.9.3:
```
libpod (master) $ /usr/bin/podman images --sort repository --format json
[
    {
        "ID": "1c35c441208254cb7c3844ba95a96485388cef9ccc0646d562c7fc026e04c807",
        "Names": [
            "docker.io/library/busybox:latest"
        ],
        "Created": "2 weeks ago",
        "Digest": "sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629f637a209",
        "Digests": [
            "sha256:95cf004f559831017cdf4628aaf1bb30133677be8702a8c5f2994629f637a209",
            "sha256:fd4a8673d0344c3a7f427fe4440d4b8dfd4fa59cfabbd9098f9eb0cb4ba905d0"
        ],
        "CreatedAt": "2020-06-02T21:19:57.279412246Z",
        "Size": 1441812,
        "ReadOnly": false,
        "History": []
    },
    {
        "ID": "a24bb4013296f61e89ba57005a7b3e52274d8edd3ae2077d04395f806b63d83e",
        "Names": [
            "docker.io/library/alpine:latest"
        ],
        "Created": "2 weeks ago",
        "Digest": "sha256:185518070891758909c9f839cf4ca393ee977ac378609f700f60a771a2dfe321",
        "Digests": [
            "sha256:185518070891758909c9f839cf4ca393ee977ac378609f700f60a771a2dfe321",
            "sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65"
        ],
        "CreatedAt": "2020-05-29T21:19:46.363518345Z",
        "Size": 5849195,
        "ReadOnly": false,
        "History": []
    }
]
```

I consider that a bug and would love to have that fixed as well. I'd guess that's one more motivation to sort the images at the caller in `func images(...)`.

**Pull request merged**